librz/bin/format/pyc/marshal.c

```diff
@@ -627,63 +648,84 @@ static pyc_object *get_set_object(RzBinPycObj *pyc, RzBuffer *buffer) {
 	return ret;
 }
 
-static pyc_object *get_ascii_object_generic(RzBuffer *buffer, ut32 size, bool interned) {
+static pyc_object *get_ascii_object_generic(RzBinPycObj *pyc, RzBuffer *buffer, ut32 size, bool interned) {
 	pyc_object *ret = NULL;
 
 	ret = RZ_NEW0(pyc_object);
 	if (!ret) {
 		return NULL;
 	}
+
+	ut64 addr = rz_buf_tell(buffer);
 	ret->type = TYPE_ASCII;
 	ret->data = get_bytes(buffer, size);
 	if (!ret->data) {
 		RZ_FREE(ret);
 	}
+
+	if (size > 0) {
+		RzBinString *string = NULL;
+		string = RZ_NEW0(RzBinString);
+		if (!string) {
+			RZ_FREE(ret);
+			return NULL;
+		}
+		string->paddr = string->vaddr = addr;
+		string->length = string->size = size;
+		string->ordinal = 0;
+		string->type = RZ_STRING_ENC_8BIT;
+		string->string = rz_str_new(ret->data);
+		RzListIter *ref_idx = rz_list_append(pyc->strings_cache, string);
+		if (!ref_idx) {
+			RZ_FREE(ret);
+			return NULL;
+		}
```

Comment:
this seems very similar to what done in the previous function, maybe extract a function and reduce duplication.